library/core/src/intrinsics/simd.rs

```diff
@@ -0,0 +1,412 @@
+//! SIMD compiler intrinsics.
+//!
+//! In this module, a "vector" is any `repr(simd)` type.
+
+extern "platform-intrinsic" {
+    /// Add two simd vectors elementwise.
+    ///
+    /// `T` must be a vector of integer or floating point primitive types.
+    pub fn simd_add<T>(x: T, y: T) -> T;
+
+    /// Subtract `rhs` from `lhs` elementwise.
+    ///
+    /// `T` must be a vector of integer or floating point primitive types.
+    pub fn simd_sub<T>(lhs: T, rhs: T) -> T;
+
+    /// Multiply two simd vectors elementwise.
+    ///
+    /// `T` must be a vector of integer or floating point primitive types.
+    pub fn simd_mul<T>(x: T, y: T) -> T;
+
+    /// Divide `lhs` by `rhs` elementwise.
+    ///
+    /// `T` must be a vector of integer or floating point primitive types.
+    ///
+    /// # Safety
+    /// For integers, `rhs` must not contain any zero elements.
+    /// Additionally for signed integers, `<int>::MIN / -1` is undefined behavior.
+    pub fn simd_div<T>(lhs: T, rhs: T) -> T;
+
+    /// Remainder of two vectors elementwise
+    ///
+    /// `T` must be a vector of integer or floating point primitive types.
+    ///
+    /// # Safety
+    /// For integers, `rhs` must not contain any zero elements.
+    /// Additionally for signed integers, `<int>::MIN / -1` is undefined behavior.
+    pub fn simd_rem<T>(lhs: T, rhs: T) -> T;
+
+    /// Elementwise vector left shift.
```

Comment:
```suggestion
    /// Elementwise vector left shift, with UB on overflow.
```
This seems worth emphasizing since the corresponding MIR operation does *not* have UB on overflow. (This also means an auto-vectorizing MIR pass , if we ever have something like that, needs to be careful.)